src/main/java/org/jabref/model/entry/KeywordList.java

```diff
@@ -53,15 +52,49 @@ public static KeywordList parse(String keywordString, Character delimiter, Chara
 
         KeywordList keywordList = new KeywordList();
 
-        StringTokenizer tok = new StringTokenizer(keywordString, delimiter.toString());
-        while (tok.hasMoreTokens()) {
-            String chain = tok.nextToken();
-            Keyword chainRoot = Keyword.of(chain.split(hierarchicalDelimiter.toString()));
-            keywordList.add(chainRoot);
+        List<String> tokens = splitRespectingEscapes(keywordString, delimiter);
```

Comment:
@Yubo-Cao What do you think about this? Maybe an ANTLR grammar would help here, too?